src/semgrep_agent/targets.py

```diff
@@ -210,21 +216,31 @@ def _get_target_files(self) -> List[Path]:
             f"| found {unit_len(walked_entries, 'file')} in the paths to be scanned",
             err=True,
         )
-        filtered: List[Path] = []
+        survived_paths: List[Path] = []
+        ignored_paths: List[Path] = []
         for elem in walked_entries:
-            if elem.survives:
-                filtered.append(elem.path)
+            paths_group = survived_paths if elem.survives else ignored_paths
+            paths_group.append(elem.path)
 
-        skipped_count = len(walked_entries) - len(filtered)
-        if skipped_count:
+        if ignored_paths:
             click.echo(
-                f"| skipping {unit_len(range(skipped_count), 'file')} based on path ignore rules",
+                f"| skipping {unit_len(ignored_paths, 'file')} based on path ignore rules",
                 err=True,
             )
+            if len(patterns) < 100:
```

Comment:
Suggestion:

- Same comment on a length guard here
- Can we add more detail to what's being skipped and why?

E.g.:
```
debug_echo(f"Ignoring files matching pattern '{p}'")
```